### PR TITLE
fix: messages rendered in correct order when message list is not inve…

### PR DESCRIPTION
**Pull request by vanGalilea**
## 🎯 Goal

Fixes issue #1505 messages will now be rendered in the correct order when message list is not inverted

## 🛠 Implementation details

Inversion of messages in the message list is done by inverted prop only.  And not being unreversed manually in hooks.

## 🎨 UI Changes
N/A
## 🧪 Testing

- use the `inverted` prop with a value of `false` on `MessageList`
- all messages should be rendered correctly, first in top, last in bottom

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Documentation is updated
- [x] New code is tested in main example apps, including all possible scenarios
  - [x] SampleApp iOS and Android
  - [x] Expo iOS and Android

### CODE REVIEW

File: package/native-package/package.json [hunk 1]
```diff
@@ -34,6 +34,7 @@
     "@types/react-native-share": "^3.3.3",
     "react-native-document-picker": "^8.1.1",
     "react-native-haptic-feedback": "^1.14.0",
-    "react-native-image-crop-picker": "^0.38.0"
+    "react-native-image-crop-picker": "^0.38.0",
+    "@stream-io/flat-list-mvcp": "0.10.2"
```
[the remark this answers is not in the document]
Comment (santhoshvai) [reply package/native-package/package.json:38]: I think its a good practice to add all peer dependencies as dev dependencies to get typing support and testing support.

File: package/src/components/MessageList/MessageList.tsx [hunk 2]
```diff
@@ -92,13 +92,12 @@ const keyExtractor = <
   StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
 >(
   item: MessageType<StreamChatGenerics>,
-) =>
```
Comment (vanGalilea) [inline package/src/components/MessageList/MessageList.tsx:95]: Sonar improvements

File: package/src/components/MessageList/MessageList.tsx [hunk 3]
```diff
@@ -1002,12 +1008,7 @@ const MessageListWithContext = <
       {!loading && (
         <>
           <View style={styles.stickyHeader}>
-            {stickyHeaderDateToRender &&
```
Comment (vanGalilea) [inline package/src/components/MessageList/MessageList.tsx:1005]: Sonar improvements

File: package/src/components/MessageList/hooks/useMessageList.ts [hunk 4]
```diff
@@ -104,9 +103,7 @@ export const useMessageList = <
       readBy: msg.id ? readData[msg.id] || false : false,
     }));
 
-  return (
-    inverted
-      ? messagesWithStylesReadByAndDateSeparator.reverse()
-      : messagesWithStylesReadByAndDateSeparator
-  ) as MessageType<StreamChatGenerics>[];
+  return [
```
Comment (vanGalilea) [inline package/src/components/MessageList/hooks/useMessageList.ts:106]: Sonar improvements

**Review by santhoshvai — APPROVED** [review]

**Pull request merged**